io/include/pcl/compression/impl/octree_pointcloud_compression.hpp

```diff
@@ -359,9 +359,16 @@ namespace pcl
           // decode differential color information
           std::vector<char>& pointDiffColorDataVector = color_coder_.getDifferentialDataVector ();
           compressed_tree_data_in_arg.read (reinterpret_cast<char*> (&point_diff_color_data_vector_size), sizeof (point_diff_color_data_vector_size));
-          pointDiffColorDataVector.resize (static_cast<std::size_t> (point_diff_color_data_vector_size));
-          compressed_color_data_len_ += entropy_coder_.decodeStreamToCharVector (compressed_tree_data_in_arg,
+          if (point_diff_color_data_vector_size == 0)
```

Comment:
Read can fail without setting the value to Zero. Maybe stick with using the IO error macro.

PS: A check on the read value does sound good (since the value shouldn't be 0 or negative)